### ISSUE TITLE
Use async reqwest to avoid blocking the UI

### ISSUE THREAD

**Issue by matze** [matze/iridium#2]
Should be possible to integrate with the GLib mainloop running the UI.

**Comment by thomaseizinger** [comment]
Do you have on concrete ideas on how you would want this to work? :)

I've never worked with GLib but my first impression was that we should probably move all the storage handling to a different thread if we want to stay away from `Arc<Mutex>` and keep the `&mut` to `Storage`.
Once it is on a different thread, we could potentially get away with just doing things synchronously and working off a job queue that is coming from a channel that the main thread feeds into.

**Comment by matze** [comment]
> Do you have on concrete ideas on how you would want this to work?

Nope, I've only had a cursory look how GLib integrates with Rust's `async` (i.e. [this release note (see Futures)](https://gtk-rs.org/blog/2019/12/15/new-release.html) and [this example](https://github.com/gtk-rs/examples/blob/master/src/bin/gio_futures_await.rs)) but have not further looked into it in particular.

I see where you are coming from but after last night's move of current state handling to the `Storage` I don't know if a thread/command queue based model would work that well now. But I am open for suggestions.

**Comment by matze** [comment]
Oops … wrong ticket.
